### PR TITLE
[CDAP-15073] fix log meta cleaner to avoid keep looping in case numbe…

### PR DESCRIPTION
**Pull request by CuriousVini**
…r of metadata files are less than batch size

ITM: https://builds.cask.co/browse/IT-ITM-158
DUT: https://builds.cask.co/browse/CDAP-DUT6927-3

### CODE REVIEW

File: cdap-watchdog/src/main/java/co/cask/cdap/logging/clean/FileMetadataCleaner.java [hunk 1]
```diff
@@ -57,14 +58,19 @@ public FileMetadataCleaner(TransactionRunner transactionRunner) {
...
       final AtomicReference<Range> range = new AtomicReference<>(Range.all());
       // stop cleaning up if there are no files to delete or if we have reached batch size
-      while (!reachedEnd && toDelete.size() < fileCleanupBatchSize) {
+      while (!reachedEnd.get() && toDelete.size() < fileCleanupBatchSize) {
         TransactionRunners.run(transactionRunner, context -> {
           StructuredTable table = context.getTable(StoreDefinition.LogFileMetaStore.LOG_FILE_META);
-          try (CloseableIterator<StructuredRow> iter = table.scan(range.get(), fileCleanupBatchSize)) {
-            while (iter.hasNext() && toDelete.size() < fileCleanupBatchSize) {
+            try (CloseableIterator<StructuredRow> iter = table.scan(range.get(), fileCleanupBatchSize)) {
+            while (toDelete.size() < fileCleanupBatchSize) {
```
Comment (poornachandra) [inline cdap-watchdog/src/main/java/co/cask/cdap/logging/clean/FileMetadataCleaner.java]: If this loop is refactored to a separate method, then the logic becomes much simpler

Comment (CuriousVini) [reply cdap-watchdog/src/main/java/co/cask/cdap/logging/clean/FileMetadataCleaner.java]: yes, that made code more readable.

File: cdap-watchdog/src/main/java/co/cask/cdap/logging/clean/FileMetadataCleaner.java [hunk 2]
```diff
@@ -100,6 +79,36 @@ public FileMetadataCleaner(TransactionRunner transactionRunner) {
...
   }
 
+  @Nullable
+  @SuppressWarnings("ConstantConditions")
+  private Range scanFilesToDelete(StructuredTable table, int fileCleanupBatchSize, long tillTime,
+                                  List<DeletedEntry> toDelete, AtomicReference<Range> range) throws IOException {
+    try (CloseableIterator<StructuredRow> iter = table.scan(range.get(), fileCleanupBatchSize)) {
+      while (iter.hasNext()) {
+        if (toDelete.size() >= fileCleanupBatchSize) {
+          return range.get();
```
Comment (poornachandra) [inline cdap-watchdog/src/main/java/co/cask/cdap/logging/clean/FileMetadataCleaner.java]: If null is returned in this case, then there is no need to check `toDelete.size() < fileCleanupBatchSize` in the enclosing while loop.

**Review by poornachandra — APPROVED** [review]
@CuriousVini Just one comment, please fix and merge

**Comment by CuriousVini** [comment]
builds are green. merging.

**Pull request merged**